src/Log.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * This file is part of OPUS. The software OPUS has been originally developed
+ * at the University of Stuttgart with funding from the German Research Net,
+ * the Federal Department of Higher Education and Research and the Ministry
+ * of Science, Research and the Arts of the State of Baden-Wuerttemberg.
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    opus4-common
+ * @package     Opus\Log
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus;
+
+use Opus\Log\LogService;
+
+class Log extends \Zend_Log
+{
+    public function __construct(\Zend_Log_Writer_Stream $writer = null)
+    {
+        parent::__construct($writer);
+    }
+
+    /**
+     * Eliminates all prior priority filters and sets a new one.
+     *
+     * @param String $priority
+     * @throws \Zend_Log_Exception
+     */
+    public function setPriority($priority)
+    {
+        if ($priority === null) {
+            $priority = LogService::DEFAULT_PRIORITY;
+        }
+
+        $level = $this->convertPriorityToInt($priority);
+
+        if ($level === null) {
+            throw new \Exception("No such priority found as " . $priority);
+        }
+
+        $this->_filters = null;
+
+        $priorityFilter = new \Zend_Log_Filter_Priority($level);
+        $this->addFilter($priorityFilter);
+    }
+
+    /**
+     * Returns the highest priority of the logger.
```

Comment:
What is the function supposed to do, looking at it from the outside? Reading this, I am asking myself, why does the logger have multiple priorities and why am I getting the highest one? What does that mean? What do we need?

Reply:
The idea behind this function is to get the level after which the messages won't be logged. So that's why returning the highest priority. Do you think we should return all the priorities/level in the form of an array?

Reply:
I think the approach isn't right, so, no I don't think we should return all levels. I think there should be only one and we should probably know it without iterating through the filters.

Reply:
Done.